### PR TITLE
Added clear-ith-bit.md

### LINKED ISSUE #1724: [Feature]: create clear-ith-bits technique used in bit manipulation
### Feature Name

Add clear-ith-bits-solution

### Feature Description

Clearing the i-th bit in a number is a common bit manipulation technique. This operation involves setting the i-th bit to 0 while leaving all other bits unchanged.

### Motivation

This will enhance bit manipulation section.

### Implementation Suggestions (Optional)

_No response_

### Feature Type

Other

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [X] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by kjl98**
### Description
Clearing the i-th bit in a number is a nifty bit manipulation technique. This involves setting the i-th bit to 0, leaving all other bits unchanged.

Fixes #1724 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/bit-manipulation/clear-ith-Bit.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Clearing the i-th bit in a number is a common bit manipulation technique. This operation involves setting the i-th bit to 0 while leaving all other bits unchanged.
+
+## Steps to Clear the i-th Bit
+**1. Create a Mask:** Create a number that has all bits set to 1 except for the i-th bit, which is set to 0.
+
+**2. AND Operation:** Perform a bitwise AND between the original number and the mask. This will clear the i-th bit while keeping all other bits unchanged.
+
+## Formula 
+> mask =∼ (1<<𝑖)
```
Comment (ajay-dhangar) [inline docs/bit-manipulation/clear-ith-Bit.md]: replace `mask =∼ (1<<𝑖)` with ``` `mask =∼ (1<<𝑖)` ```

File: docs/bit-manipulation/clear-ith-Bit.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# Key Points:
+
+## Bitwise Operators:
+
+➔ ~: Bitwise NOT (flips all bits)
+
+➔ &: Bitwise AND
+
+➔ <<: Left shift (shifts bits to the left)
```
Comment (ajay-dhangar) [inline docs/bit-manipulation/clear-ith-Bit.md]: replace `<<` with ``` `<<` ```

File: docs/bit-manipulation/clear-ith-Bit.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		return n & bitMask;
+	}
+	public static void main(String[] args) {
+		System.out.println(clearIthBit(10, 1));
+	}
+}
+```
+
+# Explanation:
+• **Create Mask:** The mask ~(1 << i) flips all bits of 1 << i. For i = 2, 1 << 2 results in 00000100, and ~00000100 results in 11111011.
```
Comment (ajay-dhangar) [inline docs/bit-manipulation/clear-ith-Bit.md]: replace `~(1 << i)` with ``` `~(1 << i)` ```, `1 << i` with ``` `1 << i` ```, and `1 << 2` with ``` `1 << 2` ```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/bit-manipulation/clear-ith-Bit.md (hunk 1 above)
Comment (kjl98) [reply docs/bit-manipulation/clear-ith-Bit.md]: Done

File: docs/bit-manipulation/clear-ith-Bit.md (hunk 2 above)
Comment (kjl98) [reply docs/bit-manipulation/clear-ith-Bit.md]: Done

File: docs/bit-manipulation/clear-ith-Bit.md (hunk 3 above)
Comment (kjl98) [reply docs/bit-manipulation/clear-ith-Bit.md]: Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by kjl98** [comment]
> Done

Thank you @ajay-dhangar for merging my changes

**Pull request merged**